backend/src/supervisors/supervisor.py

```diff
@@ -1,30 +1,86 @@
 import logging
+from src.prompts import PromptEngine
 from src.router import pick_agent
+from src.agents import DatastoreAgent, GoalAchievedAgent, UnresolvableTaskAgent, MathsAgent
+
+unresolvable_task_agent = UnresolvableTaskAgent()
+goal_achieved_agent = GoalAchievedAgent()
+list_of_agents = [ DatastoreAgent(), MathsAgent(), goal_achieved_agent, unresolvable_task_agent ]
+
+prompt_engine = PromptEngine()
+
+
+def still_solving_tasks(current_agent, iter_count):
+    agent_calling_unresolved = current_agent != "UnresolvableTaskAgent" and current_agent != "GoalAchievedAgent"
+    count_not_exceeded = iter_count < 5
+    return agent_calling_unresolved and count_not_exceeded
+
+
+def onto_next_task(best_next_step_json):
+    return best_next_step_json["current_or_next_task"].__contains__("next")
```

Comment:
From what I've read, we shouldn't be calling "magic" name methods directly like `___contains__`.

If I understand the flow correctly, this is actually a Python dictionary (??) so we can use:
```
return "next" in best_next_step_json["current_or_next_task"]
```

Reply:
This is simply checking if the router has chosen to solve the current or next task, and we're returning True if we're completing the next one and False if it's the previous.

I can see using "magic" names being bad, have adjusted to just being equivalent (from manual testing the router is quite good at returning either "current" or "next" with no variation from those strings)